modules/RTC/ortc/RTCPeerConnection.js

```diff
@@ -672,18 +1052,837 @@ export default class ortcRTCPeerConnection extends yaeti.EventTarget {
+
+            // DTLS.
+            // If 'offer' always use 'actpass'.
+            if (type === 'offer') {
+                mediaObject.setup = 'actpass';
+            } else {
+                mediaObject.setup = remoteDtlsParameters.role === 'server'
+                    ? 'active' : 'passive';
+            }
+
+            if (kind === 'audio' || kind === 'video') {
+                mediaObject.rtp = [];
+                mediaObject.rtcpFb = [];
+                mediaObject.fmtp = [];
+
+                // Array of payload types.
+                const payloads = [];
+
+                // Add codecs.
+                for (const codec of localCapabilities.codecs) {
+                    if (codec.kind && codec.kind !== kind) {
+                        continue; // eslint-disable-line no-continue
+                    }
+
+                    payloads.push(codec.preferredPayloadType);
+
+                    const rtpObject = {
+                        payload: codec.preferredPayloadType,
+                        codec: codec.name,
+                        rate: codec.clockRate
+                    };
+
+                    if (codec.numChannels > 1) {
+                        rtpObject.encoding = codec.numChannels;
+                    }
+
+                    mediaObject.rtp.push(rtpObject);
+
+                    // If codec has parameters add them into a=fmtp attributes.
+                    if (codec.parameters) {
+                        const paramFmtp = {
+                            payload: codec.preferredPayloadType,
+                            config: ''
+                        };
+
+                        for (const name of Object.keys(codec.parameters)) {
+                            /* eslint-disable max-depth */
+                            if (paramFmtp.config) {
+                                paramFmtp.config += ';';
+                            }
+                            /* eslint-enable max-depth */
+
+                            paramFmtp.config
+                                += `${name}=${codec.parameters[name]}`;
+                        }
+
+                        if (paramFmtp.config) {
+                            mediaObject.fmtp.push(paramFmtp);
+                        }
+                    }
+
+                    // Set RTCP feedback.
+                    if (codec.rtcpFeedback) {
+                        for (const fb of codec.rtcpFeedback) {
+                            mediaObject.rtcpFb.push({
+                                payload: codec.preferredPayloadType,
+                                type: fb.type,
+                                subtype: fb.parameter || undefined
+                            });
+                        }
+                    }
+                }
+
+                // If there are no codecs, set this m section as unavailable.
+                if (payloads.length === 0) {
+                    mediaObject.payloads = '7'; // Just put something.
```

Comment:
Otherwise payloads in the m line becomes an empty string. I've seen SDP parsers failing on that case. Even Firefox places a custom/random PT when disabling a m section. I'll change it to 9 (it may be any value).